src/tests/Interop/Swift/SwiftErrorHandling/SwiftErrorHandling.swift

```diff
@@ -0,0 +1,50 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+import Foundation
+
+public enum MyError: Error {
+    case runtimeError(message: String)
+}
+
+var errorMessage: String = ""
+
+public func setMyErrorMessage(bytes: UnsafePointer<UInt8>, length: Int) {
+    let data = Data(bytes: bytes, count: length)
+    errorMessage = String(data: data, encoding: .utf8)!
+}
+
+public func conditionallyThrowError(willThrow: Bool) throws -> Int {
+    if willThrow {
+        throw MyError.runtimeError(message: errorMessage)
+    } else {
+        return 42
+    }
+}
+
+public func getMyErrorMessage(from pointer: UnsafePointer<MyError>) -> UnsafePointer<UInt8>? {
+    let pointerValue = UInt(bitPattern: pointer)
+    var offsetValue: UInt
+#if arch(arm64)
+    offsetValue = 0x48
+#elseif arch(x86_64)
+    offsetValue = 0x20
+#else
+    fatalError("Unsupported architecture")
+#endif
```

Comment:
What are these constants? Please document.

Reply:
Added a comment, please take a look. Additional inspection is needed to determine what is the content of the offset.

Reply:
This seems like a part of the Swift ABI, which I would assume would be documented somewhere if we are relying upon it, no? Is this API something the runtime would provide?

Reply:
I don't believe this falls within the runtime scope. The error handling may vary depending on the binding tool implementation. One potential approach is to refactor the test to only verify its reference without inspecting the payload. I will review further the `swift_allocError` to figure out what is the exact layout of the payload.

@stephen-hawley Any thoughts on this?

Reply:
The signature for `swift_allocError` returns a `BoxPair`. Basically a pair of a pointer to the heap object for the error (`SwiftError*` in the runtime, `Error` in Swift) and a pointer to the opaque error object.

https://github.com/apple/swift/blob/98e65d015979c7b5a58a6ecf2d8598a6f7c85794/stdlib/public/runtime/ErrorObjectNative.cpp#L68

https://github.com/apple/swift/blob/98e65d015979c7b5a58a6ecf2d8598a6f7c85794/include/swift/Runtime/HeapObject.h#L95

From what I can tell error register contains the `SwiftError*` value. I believe this offset contains the size of the header, as the value of the error is tail-alllocated after the header:

https://github.com/apple/swift/blob/98e65d015979c7b5a58a6ecf2d8598a6f7c85794/stdlib/public/runtime/ErrorObject.h#L49

I think if we change this method to take an `Error` directly, we will be able to avoid using these offsets.

Reply:
Thank you for reviewing this. You are right, if the method accepts an Error directly, there is no need for these offsets.